### PR TITLE
rbac: clarify rbac empty rules means deny

### PR DESCRIPTION
**Pull request by ramaraochavali**
Commit Message: rbac: clarify rbac empty rules means deny
Additional Description:
Risk Level: N/A
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: api/envoy/extensions/filters/http/rbac/v3/rbac.proto [hunk 1]
```diff
@@ -23,6 +23,7 @@ message RBAC {
 
   // Specify the RBAC rules to be applied globally.
   // If absent, no enforcing RBAC policy will be applied.
+  // If present and is empty it is equivalent to DENY.
```
Comment (daixiang0) [inline api/envoy/extensions/filters/http/rbac/v3/rbac.proto:26]: Could change to `If present and empty, xxx`?

File: api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto [hunk 2]
```diff
@@ -390,7 +390,8 @@ message FilterStateRule {
 
   // A map of string keys to requirements. The string key is the string value
   // in the FilterState with the name specified in the *name* field above.
-  map<string, JwtRequirement> requires = 3;
+  map<string, JwtRequirement>
```
Comment (adisuissa) [inline api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto:393]: Not sure why this has changed, but it shouldn't.
Can you double check if it happens due to proto_format?

File: api/envoy/extensions/filters/network/rbac/v3/rbac.proto [hunk 3]
```diff
@@ -39,6 +39,7 @@ message RBAC {
 
   // Specify the RBAC rules to be applied globally.
   // If absent, no enforcing RBAC policy will be applied.
+  // If present and is empty it is equivalent to DENY.
```
Comment (adisuissa) [inline api/envoy/extensions/filters/network/rbac/v3/rbac.proto:42]: nit: should be the same comment here and for the http filter.

**Review by adisuissa — CHANGES_REQUESTED** [review]
Assuming this is just doc clarification (and that an empty `rules` does deny), then I think it's ok.
Please modify that the comment is the same for both the network and the http filters, and fix unnecessary change in `jwt_authn`.

File: api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto (hunk 2 above)
Comment (ramaraochavali) [reply api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto:393]: Yeah. It is proto format thing only. I will revert. Sorry. I did not notice this.

**Review by adisuissa — COMMENTED** [review]
/lgtm api

**Comment by zuercher** [comment]
Removed v2-freeze label since the change was inadvertent and reverted.

**Review by zuercher — APPROVED** [review]

**Pull request merged**